### PR TITLE
feat(webpack-cli): validate --target flag

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
Feature

**Did you add tests for your changes?**
yes
**If relevant, did you update the documentation?**
NA
**Summary**
Refers #1261 
Added `target` option validation. warn the user and set `target: 'web'` for invalid values. 

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/groups/AdvancedGroup.js [hunk 1]
```diff
@@ -29,7 +31,22 @@ class AdvancedGroup extends GroupHelper {
...
+                'async-node',
+                'node-webkit',
+                'electron-main',
+                'electron-renderer',
+                'electron-preload',
+            ];
+            if (validTargets.includes(args.target)) {
+                options.target = args.target;
+            } else {
+                logger.warn(yellow(`"${args.target}" is an invalid value for "target" option. Using "web" by default.`));
```
Comment (anshumanv) [inline packages/webpack-cli/lib/groups/AdvancedGroup.js:47]: Yep we can't assume about the platform for which they;re bundling, throwing err and displaying the possible values would be ideal

Comment (snitin315) [reply packages/webpack-cli/lib/groups/AdvancedGroup.js:47]: makes sense :+1:

**Comment by snitin315** [comment]
throw error and exit the process ?

**Comment by anshumanv** [comment]
> throw error and exit the process ?

Yes

**Comment by evilebottnawi** [comment]
Webpack already throw an error, we should add tests that we exit with same code too

**Comment by snitin315** [comment]
My take was to validate target before passing it's value to compiler , i.e, passing only valid values to compiler.

**Comment by evilebottnawi** [comment]
@snitin315 Webpack already validate options, we don't need to do double job, let's add tests on exit code and errors

**Comment by evilebottnawi** [comment]
I think we can close it, we should not validate options, because it is webpack job

**Comment by anshumanv** [comment]
yep core already throws err we can close

**Pull request closed without merging**